src/providers/npm.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, Callable, Tuple
+import requests
+import urllib.request
+import os
+import multiprocessing
+import itertools
+from semver import max_satisfying
```

Comment:
Consider for clarity
```suggestion
import semver
```